### ISSUE TITLE
Cannot load Gemma models with TGI 2.0.3

### ISSUE THREAD

**Issue by KCFindstr** [huggingface/text-generation-inference#1968]
### System Info

Running a TGI 2.0.3 docker on a 8 NVIDIA_L4 VM with 1 L4 exposed to docker.
Command:
```sh
MODEL=google/gemma-2b-it

docker run \
  -m 320G \
  --shm-size=40G \
  -e NVIDIA_VISIBLE_DEVICES=0 \
  -e MODEL_ID=$MODEL \
  -e NUM_SHARD=1 \
  -e MAX_INPUT_TOKENS=1024 \
  -e MAX_TOTAL_TOKENS=2048 \
  -e HUGGING_FACE_HUB_TOKEN=$TOKEN \
  -e TRUST_REMOTE_CODE=true \
  -e JSON_OUTPUT=true \
  -e PORT=8080 \
  -p 7080:8080 \
  --runtime=nvidia \
  $IMAGE
```

`$IMAGE` is a TGI 2.0.3 docker image.

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

1. Run the provided command with a TGI 2.0.3 image.
2. Get the following error:

> {"timestamp":"2024-05-28T21:14:13.854738Z","level":"ERROR","fields":{"message":"Shard complete standard error output:\n\nThe argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.\n/opt/conda/lib/python3.10/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n  warnings.warn(\n/opt/conda/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py:658: UserWarning: You are using a Backend <class 'text_generation_server.utils.dist.FakeGroup'> as a ProcessGroup. This usage is deprecated since PyTorch 2.0. Please use a public API of PyTorch Distributed instead.\n  warnings.warn(\nTraceback (most recent call last):\n\n  File \"/opt/conda/bin/text-generation-server\", line 8, in <module>\n    sys.exit(app())\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/cli.py\", line 90, in serve\n    server.serve(\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/server.py\", line 258, in serve\n    asyncio.run(\n\n  File \"/opt/conda/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n\n  File \"/opt/conda/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/server.py\", line 222, in serve_inner\n    model = get_model(\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/models/__init__.py\", line 441, in get_model\n    return FlashGemma(\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/models/flash_gemma.py\", line 61, in __init__\n    model = FlashGemmaForCausalLM(prefix, config, weights, causal=True)\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/models/custom_modeling/flash_gemma_modeling.py\", line 431, in __init__\n    self.embed_tokens = TensorParallelEmbedding(\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/layers/tensor_parallel.py\", line 157, in __init__\n    weight = weights.get_partial_sharded(f\"{prefix}.weight\", dim=0)\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/utils/weights.py\", line 87, in get_partial_sharded\n    filename, tensor_name = self.get_filename(tensor_name)\n\n  File \"/opt/conda/lib/python3.10/site-packages/text_generation_server/utils/weights.py\", line 63, in get_filename\n    raise RuntimeError(f\"weight {tensor_name} does not exist\")\n\nRuntimeError: weight language_model.model.embed_tokens.weight does not exist\n"},"target":"text_generation_launcher","span":{"rank":0,"name":"shard-manager"},"spans":[{"rank":0,"name":"shard-manager"}]}
> Error: ShardCannotStart

### Expected behavior

Expect the server to load and serve the model successfully.

**Comment by KCFindstr** [comment]
Is https://github.com/huggingface/text-generation-inference/pull/1950 fixing this issue?

**Comment by philschmid** [comment]
Can you try with the latest available public container `ghcr.io/huggingface/text-generation-inference:latest`? And let us know if it is still exist?

**Comment by KCFindstr** [comment]
> Can you try with the latest available public container `ghcr.io/huggingface/text-generation-inference:latest`? And let us know if it is still exist?

Thanks! The issue is fixed with this container. Is the fix going to be released with TGI 2.0.5 as I don't think https://github.com/huggingface/text-generation-inference/pull/1950 is in 2.0.4?
